apriltag_ros/src/common_functions.cpp

```diff
@@ -222,21 +235,44 @@ AprilTagDetectionArray TagDetector::detectTags (
                                   .buf = gray_image.data
   };
 
-  image_geometry::PinholeCameraModel camera_model;
-  camera_model.fromCameraInfo(camera_info);
+  // publish the header once for the tracker to know this detector exists
+  if (optical_frame_id_ != camera_info->header.frame_id)
+  {
+    optical_frame_id_ = camera_info->header.frame_id;
+    std_msgs::String optical_frame_id_msg;
+    optical_frame_id_msg.data = optical_frame_id_;
+    optical_frame_pub_.publish(optical_frame_id_msg);
+
+    // compute the pixel values for to the fov_scale_
+    fov_pixel_buffer_height_ = (camera_info->height/2) - ((camera_info->height * fov_scale_) / 2);
+    fov_pixel_buffer_width_ = (camera_info->width/2) - ((camera_info->width * fov_scale_) / 2);
+  }
+
+  camera_model_.fromCameraInfo(camera_info);
 
   // Get camera intrinsic properties for rectified image.
-  double fx = camera_model.fx(); // focal length in camera x-direction [px]
-  double fy = camera_model.fy(); // focal length in camera y-direction [px]
-  double cx = camera_model.cx(); // optical center x-coordinate [px]
-  double cy = camera_model.cy(); // optical center y-coordinate [px]
+  double fx = camera_model_.fx(); // focal length in camera x-direction [px]
+  double fy = camera_model_.fy(); // focal length in camera y-direction [px]
+  double cx = camera_model_.cx(); // optical center x-coordinate [px]
+  double cy = camera_model_.cy(); // optical center y-coordinate [px]
 
   ROS_INFO_STREAM_ONCE("Camera model: fx = " << fx << ", fy = " << fy << ", cx = " << cx << ", cy = " << cy);
 
   // Check if camera intrinsics are not available - if not the calculated
   // transforms are meaningless.
   if (fx == 0 && fy == 0) ROS_WARN_STREAM_THROTTLE(5, "fx and fy are zero. Are the camera intrinsics set?");
 
+  // return if not in fov or not in the detection range
+  if (!in_fov_ || !in_detection_range_)
+  {
+    // ROS_INFO_STREAM("Frame " << optical_frame_id_ << " is either not in FOV or detection range, skipping");
+    AprilTagDetectionArray tag_detection_array;
+    tag_detection_array.header = camera_info->header;
+    return tag_detection_array;
+  }
+
+  ROS_INFO_STREAM_THROTTLE(2.0, "Frame " << optical_frame_id_ << " is in the FOV, detecting");
```

Comment:
```suggestion
```
i assume this was just for debug

Reply:
Yeah, thought it might be worth keeping since it's nice to see what is doing the detecting but I can remove.